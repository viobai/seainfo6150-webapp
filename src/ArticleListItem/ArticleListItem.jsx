import { Link } from "react-router-dom";
import React, { useState } from "react";
import PropTypes from "prop-types";
import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton";
import ArticleImage from "../ArticleImage/ArticleImage.jsx";
import styles from "./ArticleListItem.module.css";

const ArticleListItem = (props) => {
  const [showInfo, setShow] = useState(false);
  const articleLink = "/articlelist/" + props.article.slug;

  const onClick = () => {
    setShow(!showInfo);
  };

  return (
    <li className={styles.item}>
      <article className={styles.article}>
      <ArticleImage url={props.article.image._url} title={props.article.title} />
      <Link to={articleLink} className={styles.title}><h2 className={styles.title}>{props.article.title}</h2></Link>
        {showInfo && 
          <div>
            <time className={styles.dateTime}  dateTime={props.article.timeStamp}>
              {props.article.displayDate}
            </time>
            <p>{props.article.shortText}</p>
          </div>
        }
      </article>
      
      <ArticleTextToggleButton onClick={onClick} buttonText={showInfo ? "Show Less" : "Show More"}/>
    </li>
  );
};

ArticleListItem.propTypes = {
  article: PropTypes.object.isRequired
};
export default ArticleListItem;
