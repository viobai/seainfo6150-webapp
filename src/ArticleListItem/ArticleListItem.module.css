.item {
    border: 1.5px solid rgb(210, 20, 20);
    border-radius: 10px;
    margin: 2em 2rem 1.5rem -1.5rem;
    padding: 0rem 0.8rem 0.8rem 0.8rem;
    box-shadow: -2px 10px 10px rgb(100, 100, 100), 2px 10px 10px rgb(100, 100, 100), 0 -1px 10px rgb(100, 100, 100);  
    background-image: linear-gradient(
        to right,
        rgba(200, 20, 20, 0.7),
        white 20%
    );
}

.article {
    font-family: Arial; 
    margin-bottom: 1.5rem;
}

.title {
    color: rgb(220, 10, 10);
    font-size: 32px;
}

.dateTime {
    font-weight: bold;
}