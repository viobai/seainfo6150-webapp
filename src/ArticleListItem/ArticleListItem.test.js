import React from "react";
import ArticleListItem from "./ArticleListItem";
import { fireEvent } from "@testing-library/react";

describe("ArticleListItem tests", () => {
    var article = {
        "title": "The Statue of Liberty's torch heads to new museum",
        "slug": "181122102925-statue-of-liberty-crown",
        "shortText": "The Statue of Liberty's original torch is getting a new home",
        "timeStamp": "2018-11-22T15:12:24.000Z",
        "displayDate": "November 22nd 2018, 7:12 am",
        "image": {
            "_url": "https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
        }
    };

    it("renders correctly", () => {  
        const { container } = render(<ArticleListItem article={article}/>);
        expect(container).toMatchSnapshot();
    });

    it("renders correctly after button is clicked once", () => {
        const { container, getByText } = render(<ArticleListItem article={article}/>);
        const button = getByText("Show More");
        fireEvent.click(button);
        expect(container).toMatchSnapshot();
    });

    it("renders correctly after button is clicked twice", () => {   
        const { container, getByText } = render(<ArticleListItem article={article}/>);
        const button = getByText("Show More");
        fireEvent.click(button);
        fireEvent.click(button);
        expect(container).toMatchSnapshot();
    });
});