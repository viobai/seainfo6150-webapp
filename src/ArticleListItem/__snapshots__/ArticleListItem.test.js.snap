// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <li
    class="item"
  >
    <article
      class="article"
    >
      <div
        class="wrapper"
      >
        <img
          alt="The Statue of Liberty's torch heads to new museum"
          class="img"
          src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
        />
        <a
          class="title"
          href="/articlelist/181122102925-statue-of-liberty-crown"
        >
          <h2
            class="title"
          >
            The Statue of Liberty's torch heads to new museum
          </h2>
        </a>
      </div>
      <button
        class="button"
      >
        Show More
      </button>
    </article>
  </li>
</div>
`;

exports[`ArticleListItem tests renders correctly after button is clicked once 1`] = `
<div>
  <li
    class="item"
  >
    <article
      class="article"
    >
      <div
        class="wrapper"
      >
        <img
          alt="The Statue of Liberty's torch heads to new museum"
          class="img"
          src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
        />
        <a
          class="title"
          href="/articlelist/181122102925-statue-of-liberty-crown"
        >
          <h2
            class="title"
          >
            The Statue of Liberty's torch heads to new museum
          </h2>
        </a>
      </div>
      <div
        class="desc"
      >
        <time
          class="dateTime"
          datetime="2018-11-22T15:12:24.000Z"
        >
          November 22nd 2018, 7:12 am
        </time>
        <p>
          The Statue of Liberty's original torch is getting a new home
        </p>
      </div>
      <button
        class="button"
      >
        Show Less
      </button>
    </article>
  </li>
</div>
`;

exports[`ArticleListItem tests renders correctly after button is clicked twice 1`] = `
<div>
  <li
    class="item"
  >
    <article
      class="article"
    >
      <div
        class="wrapper"
      >
        <img
          alt="The Statue of Liberty's torch heads to new museum"
          class="img"
          src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
        />
        <a
          class="title"
          href="/articlelist/181122102925-statue-of-liberty-crown"
        >
          <h2
            class="title"
          >
            The Statue of Liberty's torch heads to new museum
          </h2>
        </a>
      </div>
      <button
        class="button"
      >
        Show More
      </button>
    </article>
  </li>
</div>
`;
